### PR TITLE
EE-7827: App is unable to resolve mule-scripting

### PR DESCRIPTION
**Pull request by marianorean**

### CODE REVIEW

File: modules/spring-config/src/main/java/org/mule/runtime/config/internal/SpringXmlConfigurationBuilder.java [hunk 1]
```diff
@@ -256,19 +264,26 @@ private ArtifactAst createApplicationModel(Set<ExtensionModel> extensions,
...
 
   private AstXmlParser createMuleXmlParser(Set<ExtensionModel> extensions,
-                                           Map<String, String> artifactProperties, boolean disableXmlValidations) {
+                                           Map<String, String> artifactProperties, boolean disableXmlValidations,
+                                           MuleContext muleContext)
+      throws MuleException {
     ConfigurationPropertiesResolver propertyResolver =
         new DefaultConfigurationPropertiesResolver(empty(), new StaticConfigurationPropertiesProvider(artifactProperties));
 
+    muleContext.getInjector().inject(this);
```
Comment (IvanAndresFritzler) [inline modules/spring-config/src/main/java/org/mule/runtime/config/internal/SpringXmlConfigurationBuilder.java:273]: Use LifecycleUtils.initialiseIfNeeded

Comment (IvanAndresFritzler) [inline modules/spring-config/src/main/java/org/mule/runtime/config/internal/SpringXmlConfigurationBuilder.java:273]: Cant't we do this injection at an earlier point (I'm thinking in the doConfigure method) and avoid passing the muleContext around?

**Review by IvanAndresFritzler — CHANGES_REQUESTED** [review]

**Comment by IvanAndresFritzler** [comment]
I think that we need a test to ensure that this feature flag is kept in place.

**Comment by marianorean** [comment]
--validate

File: modules/spring-config/src/main/java/org/mule/runtime/config/internal/SpringXmlConfigurationBuilder.java [hunk 2]
```diff
@@ -421,4 +435,12 @@ public void setParentContext(MuleContext domainContext, ArtifactAst parentAst) {
   public void setComponentBuildingDefinitionRegistryFactory(ComponentBuildingDefinitionRegistryFactory componentBuildingDefinitionRegistryFactory) {
     this.componentBuildingDefinitionRegistryFactory = ofNullable(componentBuildingDefinitionRegistryFactory);
   }
+
+  AstXmlParser getParser() {
```
Comment (IvanAndresFritzler) [inline modules/spring-config/src/main/java/org/mule/runtime/config/internal/SpringXmlConfigurationBuilder.java:439]: Can this be protected?

File: modules/spring-config/src/test/java/org/mule/runtime/config/internal/SpringXmlConfigurationBuilderTestCase.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+  public void configureWithFailOnFirstError() throws ConfigurationException {
+    when(featureFlaggingService.isEnabled(ENTITY_RESOLVER_FAIL_ON_FIRST_ERROR)).thenReturn(true);
+
+    try {
+      configurationBuilder.configure(muleContext);
+    } catch (Exception ignored) {
+    }
+
+    verify(featureFlaggingService).isEnabled(ENTITY_RESOLVER_FAIL_ON_FIRST_ERROR);
+    assertThat(((DefaultAstXmlParser) configurationBuilder.getParser()).getFailStrategy(),
```
Comment (IvanAndresFritzler) [inline modules/spring-config/src/test/java/org/mule/runtime/config/internal/SpringXmlConfigurationBuilderTestCase.java:54]: can we avoid the insanceOf and just try to parse something?

**Review by IvanAndresFritzler — CHANGES_REQUESTED** [review]

File: modules/spring-config/src/test/java/org/mule/runtime/config/internal/SpringXmlConfigurationBuilderTestCase.java [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+import org.mule.runtime.api.meta.model.ExtensionModel;
+import org.mule.runtime.core.api.config.ConfigurationException;
+import org.mule.runtime.core.api.config.bootstrap.ArtifactType;
+import org.mule.runtime.core.internal.context.MuleContextWithRegistry;
+import org.mule.runtime.extension.api.dsl.syntax.resources.spi.ExtensionSchemaGenerator;
+
+import javax.inject.Inject;
+import java.util.HashMap;
+
+import static org.hamcrest.Matchers.containsString;
```
Comment (IvanAndresFritzler) [inline modules/spring-config/src/test/java/org/mule/runtime/config/internal/SpringXmlConfigurationBuilderTestCase.java:25]: static imports go first

File: modules/spring-config/src/test/java/org/mule/runtime/config/internal/SpringXmlConfigurationBuilderTestCase.java [hunk 5]
```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.config.internal;
+
+import io.qameta.allure.Issue;
```
Comment (IvanAndresFritzler) [inline modules/spring-config/src/test/java/org/mule/runtime/config/internal/SpringXmlConfigurationBuilderTestCase.java:9]: io.* and org.junit.* after org.mule.* and java/javax.*

File: modules/spring-config/src/test/java/org/mule/runtime/config/internal/SpringXmlConfigurationBuilderTestCase.java [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+import org.junit.rules.ExpectedException;
+import org.mule.runtime.api.config.FeatureFlaggingService;
+import org.mule.runtime.api.dsl.DslResolvingContext;
+import org.mule.runtime.api.meta.model.ExtensionModel;
+import org.mule.runtime.core.api.config.ConfigurationException;
+import org.mule.runtime.core.api.config.bootstrap.ArtifactType;
+import org.mule.runtime.core.internal.context.MuleContextWithRegistry;
+import org.mule.runtime.extension.api.dsl.syntax.resources.spi.ExtensionSchemaGenerator;
+
+import javax.inject.Inject;
```
Comment (IvanAndresFritzler) [inline modules/spring-config/src/test/java/org/mule/runtime/config/internal/SpringXmlConfigurationBuilderTestCase.java:22]: javax.* after java.*

**Review by IvanAndresFritzler — CHANGES_REQUESTED** [review]

File: modules/spring-config/src/main/java/org/mule/runtime/config/internal/SpringXmlConfigurationBuilder.java [hunk 7]
```diff
@@ -74,6 +77,8 @@
 import org.springframework.context.ApplicationContext;
 import org.springframework.context.ConfigurableApplicationContext;
 
+import javax.inject.Inject;
```
Comment (IvanAndresFritzler) [inline modules/spring-config/src/main/java/org/mule/runtime/config/internal/SpringXmlConfigurationBuilder.java:80]: right after org.mule.*

**Review by IvanAndresFritzler — CHANGES_REQUESTED** [review]

**Review by IvanAndresFritzler — APPROVED** [review]

**Pull request merged**